sql/core/src/main/scala/org/apache/spark/sql/jdbc/H2Dialect.scala

```diff
@@ -74,6 +77,40 @@ private[sql] object H2Dialect extends JdbcDialect {
     functionMap.clear()
   }
 
+  override def createIndex(
+      indexName: String,
+      tableIdent: Identifier,
+      columns: Array[NamedReference],
+      columnsProperties: util.Map[NamedReference, util.Map[String, String]],
+      properties: util.Map[String, String]): String = {
+    val columnList = columns.map(col => col.fieldNames.head)
+    val (indexType, _) = JdbcUtils.processIndexProperties(properties, "h2")
+    s"CREATE INDEX ${indexNameWithSchema(tableIdent, indexName)} $indexType ON " +
```

Comment:
Is the schema required here for index name? If not putting schema, does it default to the same schema as the table schema?